### DISCUSSION TITLE
Courseplay Vehicle names

### DISCUSSION THREAD

**Discussion by RayzerSharpe** [Courseplay/courseplay#7019] [Q&A] [answered]
I have some vehicles (tractors mainly) displaying some strange names and its hard to figure out sometimes which is which. What does courseplay use when deciding a vehicles name. For instance i have a Massey 8140 and its name appears as "Standard". Yes this is a mod and not a great one but are there some field missing in the xml files?  Entries in the log file appear as follows

2021-03-21 16:16 :23 [info lp15677] Standard (#2): AI driver in mode 5 starting at 19/241 waypoints (COURSEPLAY_NEXT_POINT)

**Comment by pvaiko** [comment]
We use the Giant standard getName() function:

https://github.com/Courseplay/courseplay/blob/4fd6384656168c87c7fc35d270f98d942b0ea4d5/helpers.lua#L44

**Reply by RayzerSharpe** [reply]
Have a look at the attached screen shot.. the 8140 massey is the one stuck under the combine and as you point out the Getname() is providing the name anmd number to display the slowing down and slipping messages on the left. However if you look at the list of harvesters that i want to assign to the other tractor, the massey is called 8140 and not Standard. Why the difference and any ideas why this would be the case?
![8140](https://user-images.githubusercontent.com/25461795/112400112-8e7c6c00-8cff-11eb-9249-612f29c0a5df.png)

**Reply by KITT3000** [reply]
Giants use this function:
```lua
function FSBaseMission:getVehicleName(vehicle)
	local name = vehicle:getFullName()
	name = utf8ToUpper(name)

	return name
end
```

**Reply by RayzerSharpe** [reply]
Should we use getFullName everywhere

**Comment by rohne83** [comment, marked as answer]
That FS function seems to take the vehicle name from the motor config. So at least for mods you can amend this yourself by setting a more proper name than just the default l10n_configuration_valueDefault which results in "Standard" (or whatever language you have set). I've done that for all mods I am using frequently as I was annoyed by the same

**Reply by RayzerSharpe** [reply]
aaah cheers 

On the tractor in question the setting was 

`<motorConfiguration name="$l10n_configuration_valueDefault" hp="160" price="0">`

I just changed the name and it works perfect... nice one..
